sdk/core/System.ClientModel/src/Internal/ChangeTrackingStringList.cs

```diff
@@ -26,10 +27,16 @@ public ChangeTrackingStringList(IEnumerable<string> collection)
         _tracking = false;
     }
 
+    public bool HasChanged { get; private set; }
+
     public void StartTracking() => _tracking = true;
+
     public void StopTracking() => _tracking = false;
 
-    public bool HasChanged { get; private set; }
+    public void Freeze()
```

Comment:
With this addition, maybe we would want to add `AssertNotFrozen` to this type as well.  I think it is functionally correct since we make the list read-only, but it might be more beneficial to users if they got a "you cannot modify because type is frozen" exception instead of a "you tried to modify a read-only collection" exception.